### ISSUE TITLE
Crash on OUYA with Tegra 3 CPU

### ISSUE THREAD

**Issue by kubecz3k** [godotengine/godot#1608]
**Update:** it turns out that checking an option in android export settings "Use 32 Bits View" to ON is responsible for these crashes. 

I have tried to run builds made with 1.1beta on OUYA, unfortunately it crash immediately (Platformer2D Sample). Also, I can tell it was working on 1.0.

Below is logcat output:
On the hastebin(more readable I hope): http://hastebin.com/lofivitine.mel
The same below:
```
I/ActivityManager(  326): Start proc com.android.noname for activity com.android.noname/com.android.godot.Godot: pid=4343 uid=10038 gids={1028}
I/InputReader(  326): Reconfiguring input devices.  changes=0x00000004
I/NvCpuClient(  107): Successfully bound to service
I/ActivityManager(  326): Config changed: {1.0 0mcc0mnc en_US sw540dp w540dp h935dp lrg long port television -touch -keyb/v/h -nav/h s.16}
D/GODOT   ( 4343): ** GODOT ACTIVITY CREATED HERE ***
D/dalvikvm( 4343): Trying to load lib /data/data/com.android.noname/lib/libgodot_android.so 0x42634c50
D/dalvikvm( 4343): Added shared lib /data/data/com.android.noname/lib/libgodot_android.so 0x42634c50
D/dalvikvm( 4343): No JNI_OnLoad found in /data/data/com.android.noname/lib/libgodot_android.so 0x42634c50, skipping init
D/GODOT   ( 4343): command_line is null? yes
I/godot   ( 4343): **INIT EVENT! - 0x41625718
I/godot   ( 4343): ***************** HELLO FROM JNI!!!!!!!!
I/godot   ( 4343): *******CLASS FOUND!!!
I/godot   ( 4343): STEP2, 0x1d2001f2
I/godot   ( 4343): STEP3 1113878448
I/godot   ( 4343): STEP4, 0xa2200025
I/godot   ( 4343): STEP4.5, 0x1d2001f6
I/godot   ( 4343): STEP5
I/godot   ( 4343): STEP6
I/godot   ( 4343): *******GOT METHOD _file_open ok!!
I/godot   ( 4343): *******GOT METHOD _file_get_size ok!!
I/godot   ( 4343): *******GOT METHOD _file_tell ok!!
I/godot   ( 4343): *******GOT METHOD _file_eof ok!!
I/godot   ( 4343): *******GOT METHOD _file_seek ok!!
I/godot   ( 4343): *******GOT METHOD _file_read ok!!
I/godot   ( 4343): *******GOT METHOD _file_close ok!!
I/godot   ( 4343): STEP7
I/godot   ( 4343): STEP8
I/godot   ( 4343): *******GOT METHOD _dir_open ok!!
I/godot   ( 4343): *******GOT METHOD _dir_next ok!!
I/godot   ( 4343): *******GOT METHOD _dir_close ok!!
I/godot   ( 4343): starting to attempt get methods
I/godot   ( 4343): *******GOT METHOD _init_audio ok!!
I/godot   ( 4343): *******GOT METHOD _write_buffer ok!!
I/godot   ( 4343): *******GOT METHOD _quit ok!!
I/godot   ( 4343): *******GOT METHOD _pause ok!!
I/godot   ( 4343): CMDLINE LEN 0 - APK EXPANSION I

I/godot   ( 4343): ERROR: create: Condition ' !create_func ' is true. returned: 0
I/godot   ( 4343):    At: core/os/mutex.cpp:39.

I/godot   ( 4343): test construction 5
I/godot   ( 4343): running from dir /
I/godot   ( 4343): **SETUP
I/godot   ( 4343): has res dir: 
I/godot   ( 4343): exists engine cfg? 0
I/godot   ( 4343): loaded engine.cfg
I/InputReader(  326): Reconfiguring input devices.  changes=0x00000004
I/NvCpuClient(  107): Successfully bound to service
I/ActivityManager(  326): Config changed: {1.0 0mcc0mnc en_US sw540dp w960dp h515dp lrg long land television -touch -keyb/v/h -nav/h s.17}
I/godot   ( 4343): *****SETUP OK
I/godot   ( 4343): **START
W/ActivityManager(  326): Unable to start service Intent { act=com.android.vending.billing.InAppBillingService.BIND }: not found
I/WindowManager(  326): Switching to real app window: Window{427e4080 com.android.noname/com.android.godot.Godot paused=false}
D/libEGL  ( 4343): loaded /system/lib/egl/libEGL_tegra.so
E/        ( 4343): file /data/data/com.nvidia.NvCPLSvc/files/driverlist.txt: not found!
I/        ( 4343): Attempting to load EGL implementation /system/lib//egl/libEGL_tegra_impl
I/        ( 4343): Loaded EGL implementation /system/lib//egl/libEGL_tegra_impl
D/libEGL  ( 4343): loaded /system/lib/egl/libGLESv1_CM_tegra.so
D/libEGL  ( 4343): loaded /system/lib/egl/libGLESv2_tegra.so
W/GodotView( 4343): creating OpenGL ES 2.0 context :
W/dalvikvm( 4343): threadid=10: thread exiting with uncaught exception (group=0x41634300)
E/AndroidRuntime( 4343): FATAL EXCEPTION: GLThread 152
E/AndroidRuntime( 4343): java.lang.IllegalArgumentException
E/AndroidRuntime( 4343): 	at com.google.android.gles_jni.EGLImpl._eglCreateContext(Native Method)
E/AndroidRuntime( 4343): 	at com.google.android.gles_jni.EGLImpl.eglCreateContext(EGLImpl.java:54)
E/AndroidRuntime( 4343): 	at com.android.godot.GodotView$ContextFactory.createContext(GodotView.java:397)
E/AndroidRuntime( 4343): 	at android.opengl.GLSurfaceView$EglHelper.start(GLSurfaceView.java:1030)
E/AndroidRuntime( 4343): 	at android.opengl.GLSurfaceView$GLThread.guardedRun(GLSurfaceView.java:1401)
E/AndroidRuntime( 4343): 	at android.opengl.GLSurfaceView$GLThread.run(GLSurfaceView.java:1240)
W/ActivityManager(  326):   Force finishing activity com.android.noname/com.android.godot.Godot
W/ActivityManager(  326):   Force finishing activity tv.ouya.console/.launcher.developer.MakeActivity
I/Process ( 4343): Sending signal. PID: 4343 SIG: 9
V/DropBoxReceiver(  565): Starting service for tag: data_app_crash, time: 1428061998452
I/WindowState(  326): WIN DEATH: Window{427e4080 com.android.noname/com.android.godot.Godot paused=true}
I/ActivityManager(  326): Process com.android.noname (pid 4343) has died.
D/CrashReportService(  565): Writing crash report to: /data/user/0/tv.ouya/cache/.reports/1428061998452
I/WindowState(  326): WIN DEATH: Window{42976890 SurfaceView paused=false}
```

**Comment by Krzycho666** [comment]
+1

**Comment by reduz** [comment]
Uhm no idea.. I have an ouya but since i am forced to put my cc on it,
never bothered with it

**Comment by Krzycho666** [comment]
Just put 41111111111111111 - one of those universal credit card numbers. It works like a charm ;)

**Comment by Krzycho666** [comment]
I've checked the new 1.1rc1 version just now. Some of the demos are lunching, some of them just show the blinking GodotEngine screen blinking like about 60fps. Some of them seem to work ( forgeting the fact, that 3D lags as hell ). My project ( 2D game with some shaders ) also blinks on start and does not work.
The funny thing is, that Joystick demo is blinking and SUTjoystick is running ( but it's not functional ) - it's alsmost thesame app.

**Comment by okamstudio** [comment]
maybe your android export templates are outdated?

On Mon, May 4, 2015 at 9:47 AM, Siewio <notifications@github.com> wrote:

> I've checked the new 1.1rc1 version just now. Some of the demos are
> lunching, some of them just show the blinking GodotEngine screen blinking
> like about 60fps. Some of them seem to work ( forgeting the fact, that 3D
> lags as hell ). My project ( 2D game with some shaders ) also blinks on
> start and does not work.
> The funny thing is, that Joystick demo is blinking and SUTjoystick is
> running ( but it's not functional ) - it's alsmost thesame app.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/okamstudio/godot/issues/1608#issuecomment-98697492>.
>



-- 

*OkamStudio*

**Comment by Krzycho666** [comment]
Oh. You are right. I've forgot to change templates this time. I've did it every single time before. Now it's not working at all, as it was with previous official build 1.1beta :/ So basicaly templates from v1.0stable works quite good while newer official releases doesn't at all.
Maybe just investigate when did Godot stopped working on OUYA?

**Comment by Krzycho666** [comment]
Any changes in the topic? I've tried the latest builds - with no change. Is anyone investigating on this? It seems to be something very small....

**Comment by TotCac** [comment]
Same thing on Android 5.01 (godot blink at start then crash)
If i use, simple button, it's ok, with texture button with png texture, blinking
Any news on this issue?

**Comment by Krzycho666** [comment]
Funny then. Cose I've been testing some apps on both 5.0 and 5.1, on two different devices and everything was fine ( exept one time - which is strange why it was crashin only once. I couldn't reproduce it anyway)

**Comment by TotCac** [comment]
I found a solution in another reported issue. Just restart engine. It appear to be a bug in ressources cache according to reduz.

**Comment by kubecz3k** [comment]
@TotCac Can you link to the issue? What do you mean by restarting the engine?

**Comment by TotCac** [comment]
Check this https://github.com/okamstudio/godot/issues/1926
I do not use shader, but i have blinking behaviour when exporting.
If i restart godot engine then i export to android, no more android blinking.

**Comment by kubecz3k** [comment]
Unfortunately restarting Godot engine just before exporting don't change a thing. Games still crash immediately on Ouya.

**Comment by kubecz3k** [comment]
I'm rather noob when it comes to opengl and graphic, but from my first post log it seems the engine crashes at this line:
``EGLContext context = egl.eglCreateContext(display, eglConfig, EGL10.EGL_NO_CONTEXT, use_gl2?attrib_list2:null);``
in the 
``EGLContext createContext(EGL10 egl, EGLDisplay display, EGLConfig eglConfig)`` function
with the exception ``java.lang.IllegalArgumentException``
The file with the above function: https://github.com/okamstudio/godot/blob/master/platform/android/java/src/com/android/godot/GodotView.java#L387
Maybe this will help somehow?

**Comment by reduz** [comment]
I have no idea, but Ouya should support this fine..

**Comment by akien-mga** [comment]
@kubecz3k Can you still reproduce this with 1.1-stable and/or the current master branch?

**Comment by kubecz3k** [comment]
@akien-mga yes  I have reproduced this today with current 2.0 Alpha build b2f9acb8c96aed0505cbac21661e21e4acef710f. (As well as 1.1 stable). Will change title to reflect this.

**Comment by akien-mga** [comment]
According to Miroslav on the Facebook group (and @kubecz3k on #2443), it could be specific to the Tegra 3 CPU that these devices have. Adding this to the title as it's our best clue so far.

**Comment by kubecz3k** [comment]
Kelinci Senja on facebook noticed that between Godot 1.0 and 1.1 an option in android export settings "Use 32 bits View" is now checked by default. It turns out that unchecking this option fix the problem. Not sure what we should do with this bug next, if it's possible to change this setting during runtime maybe it should be done in case of error? Should I close this issue? Maybe create fresh one?

**Comment by akien-mga** [comment]
For the reference, here is the analysis done by Kelinci Senja in https://www.facebook.com/groups/godotengine/permalink/692504434219487/?comment_id=693314364138494:

> EGLImpl.Java (Android Source) --> Last Update : Jan, 31 - 2014
> 
> GLSurfaceView.java (Android Source) --> Last update : Jun, 08 - 2015
> 
> GodotView.java (Godot Source) --> Last Update : Apr, 18 - 2015
> This file was used by the following Godot versions:
> 1.1-rc1 ( May, 01 - 2015 )
> 1.1-rc2 ( May, 05 - 2015 )
> 1.1-rc3 ( May, 17 - 2015 )
> 1.1-stable ( May, 21 - 2015 )
> 2.0-alpha
> 
> WHILE :
> 1.0-stable ( Dec, 15 - 2014 ) used GodotView.java --> Dec, 15 - 2014
> 
> Now, Jakub Grzesik said :
> "Juan Alpaca, I can confirm it ran fine in 1.0 stable. In 1.1 the bug was already present."
> 
> Based on that, my dumb analytic said :
> - EGLImpl.Java and GLSurfaceView.java did not cause the problem. It's simply because EGLImpl.java used by All versions, while GLSurfaceView.java was NOT used by 1.0-stable nor 1.1-x.
> - GodotView.java might be the problem, as both 1.0-stable and 1.1-x used different version of GodotView.java.
> 
> From GodotView.java (Dec, 15 - 2014) to GodotView.java (Apr, 18 - 2015):
> - 632 Lines ==>> 642 Lines
> - A New Feature added. It is Use 32 Bits View (Open Godot 1.1 or later - Export - Android - Screen). And this feature is ON by default.
> 
> As I do not have one of devices mentioned, my suggestion is :
> - untick or uncheck Use 32 Bits View option
> - export and test your game on Nvidia Tegra powered devices.
> 
> ![12244604_489015224638982_8686569845144003888_o](https://cloud.githubusercontent.com/assets/4701338/11318163/dbb5f74a-9046-11e5-950e-b52b208a1a33.jpg)

**Comment by reduz** [comment]
if someone with a Tegra3 device can modify GodotView.java to fallback on 16
bits when 32 bits is not available, that would be hugely appreciated

**Comment by akien-mga** [comment]
This issue should be fixed by #2862. Feel free to reopen (or ask me to reopen) if it's not the case.
